code/app/src/main/java/com/kernelcrew/moodapp/ui/FollowersFragment.java

```diff
@@ -0,0 +1,135 @@
+import androidx.navigation.Navigation;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import com.google.android.material.appbar.MaterialToolbar;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.kernelcrew.moodapp.R;
+import com.google.android.material.imageview.ShapeableImageView;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FollowersFragment extends Fragment {
+
+    private RecyclerView followersRecyclerView;
+    private FollowersAdapter adapter;
+    private List<User> followersList = new ArrayList<>();
+    private FirebaseFirestore db;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_followers, container, false);
+
+        followersRecyclerView = view.findViewById(R.id.followersRecyclerView);
+        followersRecyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));
+
+        adapter = new FollowersAdapter(followersList);
+        followersRecyclerView.setAdapter(adapter);
+
+        db = FirebaseFirestore.getInstance();
+        fetchFollowers();
+
+        BottomNavigationView bottomNavigationView = view.findViewById(R.id.bottom_navigation);
+        bottomNavigationView.setSelectedItemId(R.id.page_myProfile);
+        bottomNavigationView.setOnItemSelectedListener(item -> {
+            if (item.getItemId() == R.id.page_myProfile) {
+                Navigation.findNavController(view).navigate(R.id.myProfile);
+                return true;
+            }
+            return false;
+        });
+
+        MaterialToolbar topAppBar = view.findViewById(R.id.topAppBar);
+        topAppBar.setNavigationOnClickListener(v -> Navigation.findNavController(v).navigateUp());
+
+        return view;
+    }
+
+    private void fetchFollowers() {
+        db.collection("users").document("testUser").collection("followers")
+                .get()
+                .addOnSuccessListener(queryDocumentSnapshots -> {
+                    followersList.clear();
+                    for (DocumentSnapshot doc : queryDocumentSnapshots) {
+                        String name = doc.getId();
+                        boolean isFollowed = doc.getBoolean("isFollowed") != null && doc.getBoolean("isFollowed");
+                        followersList.add(new User(name, isFollowed));
+                    }
+                    adapter.notifyDataSetChanged();
+                })
+                .addOnFailureListener(e -> System.out.println("TEST_FIRESTORE: Failed to load followers."));
+    }
```

Comment:
We should move this to a controller class like the `MoodEventController`. That we we can add separate tests for it. Also, that way other classes can use the same code. Do note that this is probably need some massaging and you will still need to implement some of the list <-> adapter logic here.

Also,
- Does this only work for the user `testUser`?
- Why can't we do filtering in `.get()`? We can use `.where*` (eg,. https://firebase.google.com/docs/reference/android/com/google/firebase/firestore/Query#whereEqualTo(java.lang.String,java.lang.Object))